### ISSUE TITLE
SynParsing(Error("unrecognized fragment specifier"))

### ISSUE THREAD

**Issue by Wandalen** [lukaslueg/macro_railroad#25]
Hi! Thanks, but does not work on my Firefox.

![image](https://user-images.githubusercontent.com/7414255/149493156-ef9c3ec3-f8a9-4a7e-a633-2b96720c164c.png)

**Comment by lukaslueg** [comment]
Yes, this is because the parser does not recognize [`or-patterns`](https://dev-doc.rust-lang.org/stable/edition-guide/rust-2021/or-patterns-macro-rules.html) [as defined](https://doc.rust-lang.org/stable/reference/macros-by-example.html#metavariables) in the 2021 editions. We need two things, at least:

* Expand the parser to recognize `or_patterns` / `$pat_param`.
* Add a knob to the parser that controls the edition the input is parsed for, as in `2018` we do not parse `pat_param`.

We need test cases for the parser's behavior in all cases four cases (`2018`, `2021`, `pat_param`, `pat`).

**Comment by lukaslueg** [comment]
Fixed in cd2bd95f927e1c032c7908bbf9fbc4fc44545363
